Replicator/tests/ReplParams.cc

```diff
@@ -0,0 +1,129 @@
+//
+// Created by Callum Birks on 07/11/2022.
+//
+
+#include "ReplParams.hh"
+
+ReplParams::ReplParams(std::vector<C4ReplicationCollection> collections_, const AllocedDict& options,
+                       StatusCallback statusCallback, DocsEndedCallback docsEndedCallback,
+                       BlobProgressCallback blobProgressCallback, EncryptionCallback encryptionCallback,
+                       DecryptionCallback decryptionCallback, void *callbackContext_, C4SocketFactory *socketFactory_)
+{
```

Comment:
There are so many arguments in the constructor. Would you consider cascadable pattern, like
replParams.setA()
                   .setB()
...

Reply:
I simply did that so that the user can construct it like a C4ReplicatorParameters if they wish. As it inherits from C4ReplicatorParameters, they can set the other options directly; i.e. replParams.statusCallback = x.
Although, a cascadable pattern may also be something nice to implement.